### PR TITLE
backport: 'trivial' pr2

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
Based on PR1 and haven't built yet

### CODE REVIEW

File: src/compat/assumptions.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#ifndef BITCOIN_COMPAT_ASSUMPTIONS_H
+#define BITCOIN_COMPAT_ASSUMPTIONS_H
+
+#include <limits>
+
+// Assumption: We assume that the macro NDEBUG is not defined.
+// Example(s): We use assert(...) extensively with the assumption of it never
+//             being a noop at runtime.
+#if defined(NDEBUG)
+# error "Bitcoin cannot be compiled without assertions."
```
Comment (UdjinM6) [inline src/compat/assumptions.h:17]: 87cd2776e5d698822a7c88f574aa700e1ad65599:
```suggestion
# error "Dash Core cannot be compiled without assertions."
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/compat/assumptions.h (hunk 1 above)
Comment (PastaPastaPasta) [reply src/compat/assumptions.h]: Done https://github.com/dashpay/dash/compare/57572f45d40eab5388b0dff9751085cf205bbe4f..90fc61450047f81f2eb46d4d9ba57d48ad8457c9

**Review by UdjinM6 — COMMENTED** [review]
LGTM 👍  waiting for CI...

File: test/functional/wallet_basic.py [hunk 2]
```diff
@@ -325,12 +325,38 @@ def run_test(self):
...
 
+        # This will raise an exception for the invalid private key format
+        assert_raises_rpc_error(-5, "Invalid private key encoding", self.nodes[0].importprivkey, "invalid")
+
+        # This will raise an exception for importing an address with the PS2H flag
+        temp_address = self.nodes[1].getnewaddress()
+        assert_raises_rpc_error(-5, "Cannot use the p2sh flag with an address - use a script instead", self.nodes[0].importaddress, temp_address, "label", False, True)
+
+        # This will raise an exception for attempting to dump the private key of an address you do not own
+        assert_raises_rpc_error(-3, "Address does not refer to a key", self.nodes[0].dumpprivkey, temp_address)
```
Comment (UdjinM6) [inline test/functional/wallet_basic.py:343]:
```suggestion
        assert_raises_rpc_error(-4, "Private key for address %s is not known" % temp_address, self.nodes[0].dumpprivkey, temp_address)
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**